entities/storobj/storage_object.go

```diff
@@ -556,14 +554,37 @@ func (ko *Object) MarshalBinary() ([]byte, error) {
 	}
 	vectorWeightsLength := uint32(len(vectorWeights))
 
-	targetVectors, err := msgpack.Marshal(ko.Vectors)
-	if err != nil {
-		return nil, err
+	var targetVectorsOffsets []byte
+	targetVectorsOffsetsLength := uint32(0)
+	targetVectorsSegmentLength := uint32(0)
+
+	targetVectorsOffsetOrder := make([]string, 0, len(ko.Vectors))
+	if len(ko.Vectors) > 0 {
+		offset := uint32(0)
```

Comment:
Can we just use `targetVectorsSegmentLength`here and remove offset? I find the different variables difficult to follow 😅